### PR TITLE
[iOS/macOS] VPN loop detection

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/1/137249556945/task/1213962491263973
Tech Design URL: https://app.asana.com/1/137249556945/project/481882893211075/task/1213755794484490
CC:

### Description

This PR implements VPN loop detection.

### Testing Steps

1. TODO: Generate test build

### Impact and Risks

None - no user impact.

#### What could go wrong?

This could break metrics overall. We have extensive monitoring in place that will catch this, and a new pixel which will detect when loop protection is active.

### Notes to Reviewer

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Touches VPN tunnel startup/stop flow and pixel firing logic, so mistakes could impact connection behavior or observability. Changes are gated behind a feature flag (iOS) and covered by unit tests, reducing rollout risk.
> 
> **Overview**
> Adds a new `NetworkProtectionSubfeature.connectionFailureLoopDetection` / `vpnConnectionFailureLoopDetection` feature flag and wires it through iOS/macOS VPN extensions.
> 
> Introduces `ConnectionFailureLoopDetector` (persisted failure counter) and integrates it into `PacketTunnelProvider` to record on-demand start failures, reset on success/manual stop, and fire a new `.connectionFailureLoopDetected` event/pixel.
> 
> Updates iOS and macOS pixel pipelines to **suppress connection/tunnel start attempt/failure pixels** once a loop is detected, and adds a new pixel (`networkProtectionConnectionFailureLoopDetected`) to monitor when suppression is active.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 828cdde280fbe5b50f804ef148435b47110019ec. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: macOS/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift [hunk 1]
```diff
@@ -558,6 +568,8 @@ final class MacPacketTunnelProvider: PacketTunnelProvider {
 
         // MARK: -
 
+        let loopDetector = ConnectionFailureLoopDetector(store: defaults, isFeatureEnabled: true)
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply macOS/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift:571]: 🔴🔴🔴🔴🔴🔴🔴 Just a reminder to change back before merging

File: SharedPackages/VPN/Sources/VPN/Diagnostics/ConnectionFailureLoopDetector.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        if !isOnDemand {
+            resetState()
+            return false
+        }
+
+        let currentCount = (try? store.object(forKey: Keys.consecutiveFailureCount) as? Int) ?? 0
+        let count = currentCount + 1
+        try? store.set(count, forKey: Keys.consecutiveFailureCount)
+
+        return count == Self.threshold
```
Comment (diegoreymendez) [inline SharedPackages/VPN/Sources/VPN/Diagnostics/ConnectionFailureLoopDetector.swift:57]: This is checking `count == threshold` but `connectionLoopDetected` checks `counter > threshold` so I believe exactly at the threshold we fire both pixels.  Is this intentional?

Pretty minor but it might be worth considering to document this edge case with a test that checks both are fired (if intentional) or only one is.

**Review by diegoreymendez — APPROVED** [review]
Approved, as it works - but please check the comments.  Thanks!

File: macOS/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (samsymons) [reply macOS/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift:571]: Chatted on Zoom and agreed that this is fine as-is, it's low risk.
